modules/Library/library_browse.php

```diff
@@ -161,8 +161,16 @@
             ->fromPOST();
 
         foreach($activeShelves as $shelf) {
-            $libraryShelves[$shelf['gibbonLibraryShelfID']] = $itemGateway->queryItemsByShelf($shelf['gibbonLibraryShelfID'], $criteria);
+            if($shelf['type'] == 'Automatic') {
+                $itemGateway->updateShelfContents($shelf['gibbonLibraryShelfID'], $shelf['field'], $shelf['fieldValue']);
```

Comment:
It looks like this is happening every time someone loads the page. I've looked at the code and wonder what logic it's using to prevent duplicates in the shelf? 🤔

Reply:
Hi Sandra, it's my pleasure to help polish out the stuff I couldn't finish before leaving. As for the updateShelfContents gateway I tried to take duplicates into account with the:

`WHERE gibbonLibraryItemID NOT IN
            (SELECT gibbonLibraryItemID 
             FROM gibbonLibraryShelfItem
            WHERE gibbonLibraryShelfID = :libraryShelfID)`

Portion of the SQL statement, which seems to work from my testing. Although I'm not sure if two people were to simultaneously reload what the outcome would be. It is also slower if you have shelves that have content in the thousands range. If you have a couple pointers I'd appreciate the help on improving this 👍

Reply:
Aha, I see that now. Interesting. I could see where it might end up causing some load on the server if the table is constantly being written to in bulk on each page load 🤔 I wonder if the Automatic shelves are always a live query, and don't write to the table at all, instead returning their list of books for each page load. I'm happy to tinker with that idea post-merge, but curious to hear what you'd think.